sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -289,6 +295,86 @@ public void dispose() {
+                Message message = Message.Factory.create();
+                Discharge discharge = new Discharge();
+                discharge.setFail(!isCommit);
+                discharge.setTxnId(new Binary(transactionId.array()));
+                message.setBody(new AmqpValue(discharge));
+
+                final int payloadSize = messageSerializer.getSize(message);
+                final int allocationSize = payloadSize + MAX_AMQP_HEADER_SIZE_BYTES;
+
+                final byte[] bytes = new byte[allocationSize];
+
+                int encodedSize = message.encode(bytes, 0, allocationSize);
+                sendWithAck(new RetriableResponseWorkItem(bytes, encodedSize, DeliveryImpl.DEFAULT_MESSAGE_FORMAT,
+                    sink, timeout));
+            });
+        } else {
+            return RetryUtil.withRetry(
+                handler.getEndpointStates().takeUntil(state -> state == EndpointState.ACTIVE), timeout, retry)
+                .then(Mono.create(sink -> {
+                    Message message = Message.Factory.create();
+                    Discharge discharge = new Discharge();
+                    discharge.setFail(!isCommit);
+                    discharge.setTxnId(new Binary(transactionId.array()));
+                    message.setBody(new AmqpValue(discharge));
+
+                    final int payloadSize = messageSerializer.getSize(message);
+                    final int allocationSize = payloadSize + MAX_AMQP_HEADER_SIZE_BYTES;
+
+                    final byte[] bytes = new byte[allocationSize];
+
+                    int encodedSize = message.encode(bytes, 0, allocationSize);
+                    sendWithAck(new RetriableResponseWorkItem(bytes, encodedSize, DeliveryImpl.DEFAULT_MESSAGE_FORMAT,
+                        sink, timeout));
+                }));
+        }
+    }
+
+    @Override
+    public Mono<DeliveryState> createTransaction() {
+        logger.verbose("!!!! Will create new transaction ");
+        if (hasConnected.get()) {
+            return Mono.create(sink -> {
+                Message message = Message.Factory.create();
+                Declare declare = new Declare();
+                message.setBody(new AmqpValue(declare));
+
+                final int payloadSize = messageSerializer.getSize(message);
+                final int allocationSize = payloadSize + MAX_AMQP_HEADER_SIZE_BYTES;
+
+                final byte[] bytes = new byte[allocationSize];
+
+                int encodedSize = message.encode(bytes, 0, allocationSize);
+                sendWithAck(new RetriableResponseWorkItem(bytes, encodedSize, DeliveryImpl.DEFAULT_MESSAGE_FORMAT,
+                    sink, timeout));
+            });
+        } else {
+            return RetryUtil.withRetry(
+                handler.getEndpointStates().takeUntil(state -> state == EndpointState.ACTIVE), timeout, retry)
+                .then(Mono.create(sink -> {
+                    Message message = Message.Factory.create();
```

Comment:
These call the same thing down in the chain, but we generally use Proton.message(). This was probably taken from track 1 SB where they use Message.Factory.create()